microprofile/galleon-common/src/main/resources/layers/standalone/jaxrs/layer-spec.xml

```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" ?>
+<layer-spec xmlns="urn:jboss:galleon:layer-spec:1.0" name="jaxrs">
+    <dependencies>
+        <layer name="microprofile-rest-client" optional="true"/>
+    </dependencies>
+</layer-spec>
```

Comment:
@jfdenise My instinct is if we override a layer from a depended-upon FP, instead of letting Galleon combine the layer bits we are better off completely redeclaring the layer, as is done at https://github.com/wildfly/wildfly/pull/13774/commits/df965e99da9e05cbb8682eb223dfc6dace4cf0dc#diff-2fa73c0a47166ab82aea627f723525fbb1b3d1a112e07cfa018fc33bc08ddb23R7

There's some places where completely overriding is needed because the layer-spec.xml from the 'lower' FP doesn't actually get used. And it seems better to consistently do it the same way.

Doing that would also remove the need for https://github.com/wildfly/wildfly/pull/13774/commits/df965e99da9e05cbb8682eb223dfc6dace4cf0dc#diff-2fa73c0a47166ab82aea627f723525fbb1b3d1a112e07cfa018fc33bc08ddb23

It's not a big deal though; should not block merging this.